### DISCUSSION TITLE
Well, I am interested in designing a WGAN using pytorch-lightning, but feel trouble with limiting generator gradient clip and discriminator not clip.

### DISCUSSION THREAD

**Discussion by forechoandlook** [Lightning-AI/pytorch-lightning#9307] [code help: CV]
I do not want to use manual optimizer in LightningModule .

**Comment by rohitgr7** [comment]
for now, the internal gradient clipping procedures will clip gradients for both the generator and discriminator, but you can configure it yourself by disabling the one from lightning for now and configuring it manually

```py
def on_before_optimizer_step(self, trainer, pl_module, optimizer, opt_idx):
    if opt_idx == 0:  # assuming that opt_idx=0 is for generator
        params = ...
        torch.nn.utils.clip_grad_norm_(params, clip_val)
```

we will add support for it in the future :)
